src/components/TodoApp.tsx

```diff
@@ -0,0 +1,251 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useState, useContext, useEffect, useRef,
+} from 'react';
+import classNames from 'classnames';
+import { FilterStatus } from '../types/FilterStatus';
+import { TodosContext } from '../context/TodosContext';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+import { addTodo, deleteTodo, updateTodo } from '../api/todos';
+import { USER_ID } from '../variables';
+import { Todo } from '../types/Todo';
+
+export const TodoApp: React.FC = () => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+  const [toHideError, setToHideError] = useState(true);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processedTodoIds, setProcessedTodoIds] = useState<number[]>([]);
+
+  const [todosPrep, setTodos, errorMsg, setErrorMsg] = useContext(TodosContext);
+
+  const todos = todosPrep.filter((todo) => {
+    switch (filterStatus) {
+      case FilterStatus.Active:
+        return !todo.completed;
+      case FilterStatus.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const leftTodos = todosPrep.filter((todo) => !todo.completed);
+  const completedTodos = todosPrep.filter((todo) => todo.completed);
+  const leftTodosText = leftTodos.length === 1 ? 'item' : 'items';
+
+  const [allCompleated, setAllCompleated] = useState(
+    leftTodos.length === 0,
+  );
+
+  const hideError = () => {
+    setToHideError(true);
+
+    setTimeout(() => {
+      setErrorMsg('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    setAllCompleated(leftTodos.length === 0);
+  }, [leftTodos.length]);
```

Comment:
use `useMemo` and calculate based on `leftTodos` instead of using state variable